sparse/coo/core.py

```diff
@@ -504,15 +507,17 @@ def __getitem__(self, index):
 
         last_ellipsis = len(index) > 0 and index[-1] is Ellipsis
         index = normalize_index(index, self.shape)
-        if len(index) != 0 and all(not isinstance(ind, Iterable) and ind == slice(None) for ind in index):
+
+        if len(index) != 0 and all(not isinstance(ind, Iterable) and ind == slice(0, dim, 1)
+                                   for ind, dim in zip_longest(index, self.shape)):
```

Comment:
Is this change fixing an error that previously existed?  If so then is this tested somewhere?

Reply:
Not really. For the purposes of this project, I decided that `slice` objects without any `None` in them were better than those with `None` in the `start`, `stop` or step.

This (and the relevant changes in `slicing.py`) are to accommodate those changes.

Reply:
For the slice check why

    not isinstance(ind, Iterable) and ind == slice(0, dim, 1)

Rather than 

    ind == slice(0, dim, 1)

Separately, I'm curious about the use of `zip_longest`.